File: frontend/src/services/health.ts
import { http } from "./http";

export interface HealthStatus {
  status: string;
  version?: string;
  env?: string;
  connector_mode?: string;
}

export async function getHealth(): Promise<HealthStatus> {
  const { data } = await http.get<HealthStatus>("/healthz");
  return data;
}

export async function getReadiness(): Promise<HealthStatus> {
  const { data } = await http.get<HealthStatus>("/readyz");
  return data;
}

File: backend/app/api/v1/routes/health.py
"""Liveness/readiness — unauthenticated, no tenant data."""
from __future__ import annotations

from fastapi import APIRouter

from app import __version__
from app.core.config import settings

router = APIRouter()


@router.get("/healthz")
async def healthz() -> dict:
    return {"status": "ok", "version": __version__}


@router.get("/readyz")
async def readyz() -> dict:
    return {
        "status": "ready",
        "env": settings.env,
        "connector_mode": settings.connector_mode.value,
    }
